pkg/stanza/operator/input/windows/input.go

```diff
@@ -232,88 +233,77 @@ func (i *Input) read(ctx context.Context) int {
 
 // processEvent will process and send an event retrieved from windows event log.
 func (i *Input) processEvent(ctx context.Context, event Event) {
-	var providerName string // The provider name is only retrieved if needed.
```

Comment:
In principle, I think it is worth to keep this block handling exclusion prior to any call to the rendering functions: if the provider is excluded we want to avoid any rendering, since it requires XML unmarshalling that is more expensive than just checking if the provider is excluded.

Reply:
I reworked this so it remains as the first thing we do. It cleaned up the code quite a bit actually.